### ISSUE TITLE
Is there a way to return a class score vector associated with each bounding box?

### ISSUE THREAD

**Issue by peacherwu** [fizyr/keras-retinanet#692]
The function `predict_on_batch `returns ONE score with each bounding box, as in
`boxes, scores, labels = model.predict_on_batch(np.expand_dims(image, axis=0))`

When Retinanet predict the class of a box, does it compute a softmax of all classes? Is there a way I can get hold of the predicted values for all classes, like in a multi-class classification?

I tried `predict_on_batch ` and `predict `, but can't get it work.

**Comment by hgaiser** [comment]
Retinanet computes the sigmoid, not the softmax. You can do what you want, but you'd have to make it ;) the current implementation returns one box, score and label. You'd have to create a model that only does the regression and classification parts and perform the rest as you want it.

Closing this as it is not an issue with the code. Further discussions can be held on Slack.

**Comment by debunge** [comment]
@hgaiser Sorry to bring about old topic, but after a few days I still cannot figure out how to class score vector and slack seems busy with other questions. Could you please at least give a few pointers?

**Comment by HyunjiEllenPak** [comment]
@hgaiser 

Sorry to question about old issues again.

I would like to get all class scores, which are the probabilities after sigmoid function. 

How can I get it?

**Comment by HyunjiEllenPak** [comment]
In evaluation stage, having changed some codes, I finally got the class score vectors which are the sigmoid probabilities..
For the person who wants to do same thing as me, I let you know how i get it.

In keras_retinanet/models/retinanet.py/line 346,

other = model.output[2:] --> other = model.output[1:]

then, you can get the probabilities from

boxes, scores, labels, probs = model.predict_on_batch(np.expand_dims(image, axis=0))[:4]
